### PR TITLE
 Docstrings for `custom_gates.py`

### PR DESCRIPTION
**Pull request by bharat-thotakura**
Added more complete docstrings for custom_gates in qss.

### CODE REVIEW

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py [hunk 1]
```diff
@@ -25,6 +19,17 @@ def __init__(
         sandwich_rx_rads: float = 0,
         label: Optional[str] = None,
     ) -> None:
+        """Initializes an AceCR gate.
+
+        Args:
+            rads: Angle of rotation for CR gate (i.e., twice the angle for each echoed half-CR).
+            sandwich_rx_rads: Angle of rotation for an rx gate applied to the "X side"
+                simultaneously with the X gate on the "Z side".
+            label: An optional label for the constructed Gate. Defaults to None.
```
Comment (knsmith) [inline qiskit-superstaq/qiskit_superstaq/custom_gates.py:28]:
```suggestion
            label: An optional label for the constructed gate. Defaults to None.
```

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py [hunk 2]
```diff
@@ -106,16 +114,20 @@ class ZZSwapGate(qiskit.circuit.Gate):
...
 
     def __init__(self, theta: float, label: Optional[str] = None) -> None:
-        """Args:
-        theta: ZZ-interaction angle in radians
-        label: an optional label for the constructed Gate
+        """Initalizes a ZZ-SWAP gate.
+
+        Args:
+            theta: The ZZ-interaction angle in radians.
+            label: An optional label for the constructed Gate. Defaults to None.
```
Comment (knsmith) [inline qiskit-superstaq/qiskit_superstaq/custom_gates.py:121]:
```suggestion
            label: An optional label for the constructed gate. Defaults to None.
```

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py [hunk 3]
```diff
@@ -250,6 +297,16 @@ def control(
         label: Optional[str] = None,
         ctrl_state: Optional[Union[str, int]] = None,
     ) -> qiskit.circuit.ControlledGate:
+        """Method to return a controlled version of the Gate.
```
Comment (knsmith) [inline qiskit-superstaq/qiskit_superstaq/custom_gates.py:300]:
```suggestion
        """Method to return a controlled version of the gate.
```

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py [hunk 4]
```diff
@@ -263,19 +320,29 @@ def __str__(self) -> str:
...
 
-class iXdgGate(qiskit.circuit.Gate):  # pylint: disable=missing-class-docstring
+class iXdgGate(qiskit.circuit.Gate):
+    r"""The conjugate transpose of the `iXGate` (:math:`iXGate^{\dagger} = iXdgGate`)."""
+
     def __init__(self, label: Optional[str] = None) -> None:
+        """Initializes an iXdgGate.
+
+        Args:
+            label: An optional label for the constructed Gate. Defaults to None.
```
Comment (knsmith) [inline qiskit-superstaq/qiskit_superstaq/custom_gates.py:330]:
```suggestion
            label: An optional label for the constructed gate. Defaults to None.
```

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py [hunk 5]
```diff
@@ -284,6 +351,16 @@ def control(
         label: Optional[str] = None,
         ctrl_state: Optional[Union[str, int]] = None,
     ) -> qiskit.circuit.ControlledGate:
+        """Method to return a controlled version of the Gate.
```
Comment (knsmith) [inline qiskit-superstaq/qiskit_superstaq/custom_gates.py:354]:
```suggestion
        """Method to return a controlled version of the gate.
```

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py [hunk 6]
```diff
@@ -351,8 +460,15 @@ def __str__(self) -> str:
...
 
-class AQTiCCXGate(iCCXGate):  # pylint: disable=missing-class-docstring
+class AQTiCCXGate(iCCXGate):
+    """A subclass of the iCCXGate for AQT where the control state is "00"."""
+
     def __init__(self, label: Optional[str] = None) -> None:
+        """Initializes an AQTiCCXGate.
+
+        Args:
+            label: An optional label for the constructed Gate. Defaults to None.
```
Comment (knsmith) [inline qiskit-superstaq/qiskit_superstaq/custom_gates.py:470]:
```suggestion
            label: An optional label for the constructed gate. Defaults to None.
```

**Comment by bharat-thotakura** [comment]
> The docstrings go back and forth with using `gate` and `Gate` when used alone sentences. Either one is fine, but I'm not sure which one is better to use consistently! Other than that, LG2M!

Yeah, the previous docstrings used `Gate` so I kept it but I've gone with the lowercase `gate` for everything since it fits better in more cases.

**Comment by knsmith** [comment]
> > The docstrings go back and forth with using `gate` and `Gate` when used alone sentences. Either one is fine, but I'm not sure which one is better to use consistently! Other than that, LG2M!
> 
> Yeah, the previous docstrings used `Gate` so I kept it but I've gone with the lowercase `gate` for everything since it fits better in more cases.

Awesome :) Probably was a bit of a nit, but I guess it doesn't hurt to be consistent! Looks great otherwise :)

**Pull request merged**